packages/text-editor/src/components/MentionPopup.svelte

```diff
@@ -0,0 +1,173 @@
+<!--
+// Copyright © 2020, 2021 Anticrm Platform Contributors.
+// Copyright © 2021 Hardcore Engineering Inc.
+// 
+// Licensed under the Eclipse Public License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License. You may
+// obtain a copy of the License at https://www.eclipse.org/legal/epl-2.0
+// 
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// 
+// See the License for the specific language governing permissions and
+// limitations under the License.
+-->
+<script lang="ts">
+  import { getResource } from '@hcengineering/platform'
+  import { createFocusManager, FocusHandler, Label, ListView, resizeObserver } from '@hcengineering/ui'
+  import { createEventDispatcher } from 'svelte'
+  import presentation, {
+    getClient,
+    hasResource,
+    ObjectSearchCategory,
+    ObjectSearchResult
+  } from '@hcengineering/presentation'
+
+  export let query: string = ''
+  export let maxItemsPerCategory = 3
+
+  type SearchSection = { category: ObjectSearchCategory; items: ObjectSearchResult[] }
+  type SearchItem = { num: number; item: ObjectSearchResult; category: ObjectSearchCategory }
+
+  let items: SearchItem[] = []
+  let categories: ObjectSearchCategory[] = []
+
+  const client = getClient()
+
+  client.findAll(presentation.class.ObjectSearchCategory, { context: 'mention' }).then((r) => {
+    categories = r.filter((it) => hasResource(it.query))
+    updateItems(query)
+  })
+
+  const dispatch = createEventDispatcher()
+
+  let list: ListView
+  let scrollContainer: HTMLElement
+  let selection = 0
+
+  function dispatchItem (item: ObjectSearchResult): void {
+    dispatch('close', item)
+  }
+
+  export function onKeyDown (key: KeyboardEvent): boolean {
```

Comment:
I have a problem with closing this popup when no results found. Maybe we need to add better handling for Esc, Enter, Tab, RightArrow buttons. Currently only adding double spaces works.